### PR TITLE
Initialize pointer before using it.

### PR DESCRIPTION
**Pull request by skasperski**

### CODE REVIEW

File: src/orocos_cpp.cpp [hunk 1]
```diff
@@ -125,9 +125,11 @@ bool OrocosCpp::initialize(const OrocosCppConfig& config, bool quiet)
     //Init Bundle
     if(config.init_bundle){
         if(!quiet) std::cout << "\nInitializing Bundle.." << std::endl;
+        bundle.reset(new Bundle);
         st = bundle->initialize(config.load_task_configs);
         if(!st){
             std::cerr << "Error during initialization of Bundle" << std::endl;
+            bundle.reset();
```
Comment (skasperski) [inline src/orocos_cpp.cpp:132]: Apparently there is no other way to ask if there is an active bundle then checking this public member object.

**Comment by saarnold** [comment]
Not sure why this is a public member here in the first place, since it appears that Bundle is used as a Singleton in most cases. Can we remove it here an use the Singleton interface instead?

File: src/orocos_cpp.cpp (hunk 1 above)
Comment (maltewi) [reply src/orocos_cpp.cpp:132]: If the variable should be public or not is one thing, but that it need to be initialized is right for sure.

Also it is right that is should happen here. The OrocosCPP class is thought of to be the single point of initialization (for simplification). If bundles are supposed to be used, they are initialized here.

**Comment by maltewi** [comment]
The OrocosCPP class is thought of to be the single point of initialization (for simplification and clarity). If bundles are configured to be used, they are initialized here and there can (and IMO should) be a direct ownership. So, there's no need for a singleton instantiation in this case.

**Review by maltewi — APPROVED** [review]
see comments in conversation thread

**Pull request merged**